### PR TITLE
Added l2 single/batch timelock stage

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src-ts/proposalPipeline.ts [hunk 1]
```diff
@@ -94,7 +96,7 @@ export class StageTracker extends EventEmitter {
     while (polling) {
       try {
         const status = await this.stage.status();
-        if (currentStatus !== status) {
+        // if (currentStatus !== status) {
```
Comment (DZGoldman) [inline src-ts/proposalPipeline.ts:99]: remove?

File: src-ts/proposalStage.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import {
 import { Outbox__factory } from "@arbitrum/sdk/dist/lib/abi/factories/Outbox__factory";
 import { L2ToL1TxEvent as NitroL2ToL1TransactionEvent } from "@arbitrum/sdk/dist/lib/abi/ArbSys";
 import { OutBoxTransactionExecutedEvent } from "@arbitrum/sdk/dist/lib/abi/Outbox";
-import { TransactionReceipt } from "@ethersproject/providers";
+import { Log, TransactionReceipt } from "@ethersproject/providers";
```
Comment (DZGoldman) [inline src-ts/proposalStage.ts:15]: unused

File: src-ts/proposalStage.ts [hunk 3]
```diff
@@ -21,8 +21,10 @@ import {
   L2ArbitrumGovernor__factory,
   GovernorUpgradeable__factory,
   SecurityCouncilNomineeElectionGovernor__factory,
+  ArbSys__factory,
 } from "../typechain-types";
 import { Inbox__factory } from "@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory";
+import { RollupUserLogic__factory } from "@arbitrum/sdk/dist/lib/abi/factories/RollupUserLogic__factory";
```
Comment (DZGoldman) [inline src-ts/proposalStage.ts:27]: unused

**Review by DZGoldman — APPROVED** [review]

File: src-ts/proposalPipeline.ts (hunk 1 above)
Comment (yahgwai) [reply src-ts/proposalPipeline.ts]: woops, good catch

File: src-ts/proposalStage.ts (hunk 2 above)
Comment (yahgwai) [reply src-ts/proposalStage.ts]: updated

File: src-ts/proposalStage.ts (hunk 3 above)
Comment (yahgwai) [reply src-ts/proposalStage.ts]: updated

**Pull request merged**